Add environment variable to force no weights_only load

mikaylagawarecki opened the pull request:
In preparation for `weights_only` flip, if users don't have access to the `torch.load` call

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138225

mikaylagawarecki committed:
Add environment variable to force no weights_only load

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"

In preparation for `weights_only` flip, if users don't have access to the `torch.load` call




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"

In preparation for `weights_only` flip, if users don't have access to the `torch.load` call




[ghstack-poisoned]

albanD commented on docs/source/miscellaneous_environment_variables.rst:
```diff
@@ -8,7 +8,10 @@ Miscellaneous Environment Variables
     - Description
   * - ``TORCH_FORCE_WEIGHTS_ONLY_LOAD``
-    - If set to [``1``, ``y``, ``yes``, ``true``], the torch.load will use ``weight_only=True``. For more documentation on this, see :func:`torch.load`.
+    - If set to [``1``, ``y``, ``yes``, ``true``], the torch.load will use ``weights_only=True``. For more documentation on this, see :func:`torch.load`.
```
nit: since this is different from the other one, I think it's worth explicitly mentioning that this one will change the value even if the user explicitly provided False.

albanD commented on torch/serialization.py:
```diff
@@ -1277,20 +1277,37 @@ def _get_wo_message(message: str) -> str:
+        weights_only = True
+    elif force_no_weights_only_load:
+        if weights_only is None:
+            warnings.warn(
```
Why this warning?

mikaylagawarecki replied:
removed

albanD reviewed (commented):
Sounds good.

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"

In preparation for `weights_only` flip, if users don't have access to the `torch.load` call




[ghstack-poisoned]

albanD reviewed (approved):
Sounds good!

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"

In preparation for `weights_only` flip, if users don't have access to the `torch.load` call




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorch merge -r

mikaylagawarecki commented:
@pytorchbot merge -r

mikaylagawarecki committed:
Update

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add environment variable to force no weights_only load"

In preparation for `weights_only` flip, if users don't have access to the `torch.load` call




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge -r

mikaylagawarecki committed:
Update

[ghstack-poisoned]
